src/components/ShoppingList/ShoppingList.js

```diff
@@ -40,17 +47,44 @@ function ShoppingList({ listId }) {
       );
     } else {
       return (
-        <ul className="shopping-list__list list-reset">
-          {listItems.docs.map((doc) => (
-            <ShoppingListItem
-              key={doc.id}
-              listId={listId}
-              itemId={doc.id}
-              item={doc.data()}
-              handleCheck={handleCheck}
+        <div>
+          <div className="filter">
+            <label htmlFor="filterInput" className="filter__label label">
+              Filter items
+            </label>
+            <input
+              type="text"
+              id="filterInput"
+              name="filterInput"
+              value={filter}
+              onChange={handleInput}
+              className="filter__text-field text-field"
             />
-          ))}
-        </ul>
+            <button
+              type="button"
+              aria-label="clear"
+              className="filter__button"
+              onClick={() => setFilter('')}
+            >
+              Clear Filter
+            </button>
+          </div>
+          <ul className="shopping-list__list list-reset">
+            {listItems.docs
+              .filter((doc) =>
+                new RegExp(filter, 'i').test(doc.data().itemName),
+              )
```

Comment:
I typically see the use of array `.include()` method but this is also very nice!

Reply:
@yenly The regex came about as result of a mob session idea to make the filter case-insensitive.